torch/distributed/_tensor/device_mesh.py

```diff
@@ -110,13 +110,15 @@ def __init__(
         mesh: Union[torch.Tensor, "ArrayLike"],
         *,
         _init_process_groups: bool = True,
+        _validate_mesh: bool = True,
     ) -> None:
         self.device_type = device_type
         self.mesh = (
             mesh.detach()
             if isinstance(mesh, torch.Tensor)
             else torch.tensor(mesh, dtype=torch.int)
         )
+        self._validate_mesh = _validate_mesh
```

Comment:
I think we shouldn't make this an attribute for device mesh unless there's a specific reason, you pass pass it down to `init_process_groups` instead.